meson.build

```diff
@@ -45,6 +45,63 @@ if get_option('enable_debugger') == 'heavy'
   conf_data.set10('C_HEAVY_DEBUG', true)
 endif
 
+# prepare configuration for dynrec core
+#
+# host_machine is a machine we're building for
+#
+# In code we have code paths for two additional architectures:
+#
+# - ARMV4LE - autoconf used to set it when "armv6l" was reported, presumably
+#   it covers everything up to ARMv7.  Notably, Raspberry Pi 1 is using ARMv6.
+#   This arch was marked as NOT handling unaligned memory.
+# - MIPSEL - which used to be called MIPSEL32, and seems to be never hooked
+#   into the autoconf buildsystem (but it was used by several DOSBox ports)
+#
+selected_core = get_option('dynamic_core')
+host_cpu = host_machine.cpu_family()
+conf_data.set('C_TARGETCPU', 'UNKNOWN')
+cpu_handles_unaligned_memory = false
+
+core_selection = [
+  # cpu         selected_core       dynrec_define     C_TARGETCPU
+  [ 'x86_64',  ['auto', 'dyn-x86'], 'C_DYNAMIC_X86',  'X86_64'    ],
+  [ 'x86',     ['auto', 'dyn-x86'], 'C_DYNAMIC_X86',  'X86'       ],
+  [ 'x86_64',  ['dynrec'],          'C_DYNREC',       'X86_64'    ],
+  [ 'x86',     ['dynrec'],          'C_DYNREC',       'X86'       ],
+  [ 'aarch64', ['auto', 'dynrec'],  'C_DYNREC',       'ARMV8LE'   ],
+  [ 'arm',     ['auto', 'dynrec'],  'C_DYNREC',       'ARMV7LE'   ],
+  [ 'ppc64',   ['auto', 'dynrec'],  'C_DYNREC',       'PPC64LE'   ], # stable name
+  [ 'ppc64le', ['auto', 'dynrec'],  'C_DYNREC',       'PPC64LE'   ], # unstable name
+  [ 'ppc',     ['auto', 'dynrec'],  'C_DYNREC',       'POWERPC'   ],
+]
+
+foreach line : core_selection 
+  cpu = line[0]
+  opts_for_arch = line[1]
+  dynrec_define = line[2]
+  target_cpu = line[3]
+
+  if (host_cpu == cpu) and opts_for_arch.contains(selected_core)
+    conf_data.set('C_TARGETCPU', target_cpu)
+    conf_data.set(dynrec_define, 1)
+    cpu_handles_unaligned_memory = true
+  endif
+endforeach
+
+conf_data.set10('C_UNALIGNED_MEMORY', cpu_handles_unaligned_memory)
+
+target_cpu = conf_data.get('C_TARGETCPU')
```

Comment:
If the above for-loop search for `host_cpu == cpu` wasn't found, then `conf_data.get('C_TARGETCPU')` will be empty.

Maybe this doesn't matter though? If conf_data.get('C_TARGETCPU') is empty and neither DYNAMIC* or DYNREC* are available, then we just get vanilla "normal-core only" build?

Reply:
> Maybe this doesn't matter though? If conf_data.get('C_TARGETCPU') is empty and neither DYNAMIC* or DYNREC* are available, then we just get vanilla "normal-core only" build?

That's correct :)

When variable is not defined, then `#mesondefine C_TARGETCPU` in config.h.in file will evaluate to:

    /* #undef C_TARGETCPU */

And this define is used exclusively inside `src/cpu/`.

Majority of code in this PR is relevant to `src/cpu/` directory only… I am thinking about moving the logic in there to keep our top-level meson file a bit cleaner.

Reply:
I moved the code and reorganized setting these defines in conf_data - should be more readable now.